nixos/tests/pwquality.nix

```diff
@@ -0,0 +1,131 @@
+{ lib, ... }:
+let
+  settings = {
+    minlen = 10;
+
+    # require each class: lowercase uppercase digit and symbol/other
+    minclass = 4;
+
+    badwords = [
+      "foobar"
+      "hunter42"
+      "password"
+    ];
+
+    enforce_for_root = true;
+  };
+in
+{
+  name = "pam-pwquality";
```

Comment:
You could move this test to `nixos/tests/pam/` for findability.

Reply:
It's 99% testing pam focused functionality still so yeah I'll move it back